packages/widget/src/hooks/useBridgeSelectionData.ts

```diff
@@ -0,0 +1,90 @@
+import { useState, useEffect } from 'react'
+
+import { CHAINS_ARRAY } from '@/constants/chains'
+
+export const useBridgeSelectionData = () => {
+  const [originChain, setOriginChain] = useState('')
+  const [originToken, setOriginToken] = useState('')
+  const [destinationChain, setDestinationChain] = useState('')
+  const [destinationToken, setDestinationToken] = useState('')
+
+  useEffect(() => {
+    let originChainSelect = null
+    let destinationChainSelect = null
+
+    let originTokenSelect = null
+    let destinationTokenSelect = null
+
+    const getChainValues = () => {
+      if (
+        originChainSelect &&
+        destinationChainSelect &&
+        originTokenSelect &&
+        destinationTokenSelect
+      ) {
+        const originChainName = originChainSelect.textContent?.trim() || ''
+        const destinationChainName =
+          destinationChainSelect.textContent?.trim() || ''
+
+        const originTokenSymbol = originTokenSelect.textContent?.trim() || ''
+        const destinationTokenSymbol =
+          destinationTokenSelect.textContent?.trim() || ''
+
+        setOriginChain(originChainName)
+        setOriginToken(originTokenSymbol === 'Token' ? null : originTokenSymbol)
+        setDestinationChain(destinationChainName)
+        setDestinationToken(
+          destinationTokenSymbol === 'Token' ? null : destinationTokenSymbol
+        )
+      }
+    }
+
+    const checkElements = () => {
+      originChainSelect = document.getElementById('origin-chain-select')
+      destinationChainSelect = document.getElementById(
+        'destination-chain-select'
+      )
+      originTokenSelect = document.getElementById('origin-token-select')
+      destinationTokenSelect = document.getElementById(
+        'destination-token-select'
+      )
+
+      if (
+        originChainSelect &&
+        destinationChainSelect &&
+        originTokenSelect &&
+        destinationTokenSelect
+      ) {
+        // Get initial values when elements are available
+        getChainValues()
+
+        const observer = new MutationObserver(getChainValues)
+        const config = { childList: true, characterData: true, subtree: true }
+
+        observer.observe(originChainSelect, config)
+        observer.observe(destinationChainSelect, config)
+        observer.observe(originTokenSelect, config)
+        observer.observe(destinationTokenSelect, config)
+
+        return () => {
+          observer.disconnect()
+        }
+      } else {
+        // If elements are not available, check again after a short delay
+        const timerId = setTimeout(checkElements, 100)
+        return () => clearTimeout(timerId)
+      }
+    }
+
+    checkElements()
+  }, [])
```

Comment:
Yeah ideally we'd use redux values to provide this, but unfortunately bc this is an npm package and not a full React app (so we don't have an App.js file), it would require a rewrite of organizational code, which don't feel is necessary at the moment. Can revisit again later if we need to.